Adopt more smart pointers in Page, Quirks, and /page/csp

RupinMittal opened the pull request:
#### 0a03566a8de562433f5d5c56700dafecca3e5c64
<pre>
Adopt more smart pointers in Page, Quirks, and /page/csp
<a href="https://bugs.webkit.org/show_bug.cgi?id=297809">https://bugs.webkit.org/show_bug.cgi?id=297809</a>
<a href="https://rdar.apple.com/158978230">rdar://158978230</a>

Reviewed by Chris Dumez.

* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::protectedMainFrameOrigin const):
(WebCore::Page::logMediaDiagnosticMessage const):
(WebCore::Page::setPageScaleFactor):
(WebCore::Page::scheduleRenderingUpdateInternal):
(WebCore::Page::doAfterUpdateRendering):
(WebCore::Page::bestMediaElementForRemoteControls):
(WebCore::Page::setActivityState):
(WebCore::Page::setUnderPageBackgroundColorOverride):
(WebCore::Page::addRelevantRepaintedObject):
(WebCore::Page::addRelevantUnpaintedObject):
(WebCore::Page::logNavigation):
(WebCore::Page::checkedRenderingUpdateScheduler):
(WebCore::Page::recomputeTextAutoSizingInAllFrames):
(WebCore::Page::updateElementsWithTextRecognitionResults):
(WebCore::Page::deleteRemovedNodesAndDetachedRenderers):
(WebCore::Page::updateFixedContainerEdges):
(WebCore::Page::updateActiveNowPlayingSessionNow):
(WebCore::Page::requiresScriptTrackingPrivacyProtections const):
(WebCore::Page::requiresUserGestureForAudioPlayback const):
(WebCore::Page::requiresUserGestureForVideoPlayback const):
(WebCore::Page::protectedMediaSessionManager):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::allowedAutoplayQuirks):
(WebCore::Quirks::isEmbedDomain const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::triggerOptionalStorageAccessIframeQuirk const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::responseHeaders const):
(WebCore::ContentSecurityPolicy::applyPolicyToScriptExecutionContext):
(WebCore::ContentSecurityPolicy::allowJavaScriptURLs const):
(WebCore::ContentSecurityPolicy::allowInlineEventHandlers const):
(WebCore::ContentSecurityPolicy::allowNonParserInsertedScripts const):
(WebCore::ContentSecurityPolicy::allowInlineScript const):
(WebCore::ContentSecurityPolicy::allowInlineStyle const):
(WebCore::ContentSecurityPolicy::allowEval const):
(WebCore::ContentSecurityPolicy::allowFrameAncestors const):
(WebCore::ContentSecurityPolicy::allowPluginType const):
(WebCore::ContentSecurityPolicy::allowObjectFromSource const):
(WebCore::ContentSecurityPolicy::allowChildFrameFromSource const):
(WebCore::ContentSecurityPolicy::allowResourceFromSource const):
(WebCore::ContentSecurityPolicy::allowWorkerFromSource const):
(WebCore::ContentSecurityPolicy::allowScriptFromSource const):
(WebCore::ContentSecurityPolicy::allowStyleFromSource const):
(WebCore::ContentSecurityPolicy::allowConnectToSource const):
(WebCore::ContentSecurityPolicy::allowBaseURI const):
(WebCore::ContentSecurityPolicy::allowTrustedTypesPolicy const):
(WebCore::ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup const):
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::checkedCompositor):
* Source/WebCore/rendering/RenderView.h:

Canonical link: <a href="https://commits.webkit.org/300009@main">https://commits.webkit.org/300009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9cdf5274bb31c6c746f429ab809da461cfed3122

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120732 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40426 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31078 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127127 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72807 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d69f70ad-bd7c-4d03-9de6-145fd99ce604) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122608 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41123 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49003 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91702 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60948 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/584bc932-55ef-4f1c-ae0f-c082b2cf9c95) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123684 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32830 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108215 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72250 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/72db11ac-84c2-4e77-a0b0-6b869d73048e) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31859 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26324 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70731 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102321 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26505 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129993 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47653 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36179 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100314 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48021 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104398 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100153 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25461 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45601 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23638 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44313 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47515 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53220 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46984 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50330 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48670 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
```diff
@@ -822,7 +823,7 @@ bool ContentSecurityPolicy::requireTrustedTypesForSinkGroup(const String& sinkGr
 bool ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup(const String& stringContext, const String& sink, const String& sinkGroup, StringView source) const
 {
-    return std::ranges::all_of(m_policies, [&](auto& policy) {
+    return std::ranges::all_of(m_policies, [checkedThis = CheckedRef { *this }, &stringContext, &sink, &sinkGroup, &source](auto& policy) {
```
Hm... arguments to `std::ranges::all_of` should probably be treated as `NOESCAPE`. We could either hard-code it into the checker or wrap it in a WTF function which with `NOESCAPE` annotation.

rniwa replied:
Actually, `std::ranges::all_of` is a template function with inline definition so we shouldn't have to add annotation. Will look into it. Meanwhile, please revert this change and file a radar on it.

RupinMittal replied:
Filed rdar://160389651 and sent you, and I've reverted this change here.

RupinMittal committed:
Adopt more smart pointers in Page, Quirks, and /page/csp
https://bugs.webkit.org/show_bug.cgi?id=297809
rdar://158978230

Reviewed by Chris Dumez.

* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::protectedMainFrameOrigin const):
(WebCore::Page::logMediaDiagnosticMessage const):
(WebCore::Page::setPageScaleFactor):
(WebCore::Page::scheduleRenderingUpdateInternal):
(WebCore::Page::doAfterUpdateRendering):
(WebCore::Page::bestMediaElementForRemoteControls):
(WebCore::Page::setActivityState):
(WebCore::Page::setUnderPageBackgroundColorOverride):
(WebCore::Page::addRelevantRepaintedObject):
(WebCore::Page::addRelevantUnpaintedObject):
(WebCore::Page::logNavigation):
(WebCore::Page::checkedRenderingUpdateScheduler):
(WebCore::Page::recomputeTextAutoSizingInAllFrames):
(WebCore::Page::updateElementsWithTextRecognitionResults):
(WebCore::Page::deleteRemovedNodesAndDetachedRenderers):
(WebCore::Page::updateFixedContainerEdges):
(WebCore::Page::updateActiveNowPlayingSessionNow):
(WebCore::Page::requiresScriptTrackingPrivacyProtections const):
(WebCore::Page::requiresUserGestureForAudioPlayback const):
(WebCore::Page::requiresUserGestureForVideoPlayback const):
(WebCore::Page::protectedMediaSessionManager):
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Quirks.cpp:
(WebCore::allowedAutoplayQuirks):
(WebCore::Quirks::isEmbedDomain const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::triggerOptionalStorageAccessIframeQuirk const):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::responseHeaders const):
(WebCore::ContentSecurityPolicy::applyPolicyToScriptExecutionContext):
(WebCore::ContentSecurityPolicy::allowJavaScriptURLs const):
(WebCore::ContentSecurityPolicy::allowInlineEventHandlers const):
(WebCore::ContentSecurityPolicy::allowNonParserInsertedScripts const):
(WebCore::ContentSecurityPolicy::allowInlineScript const):
(WebCore::ContentSecurityPolicy::allowInlineStyle const):
(WebCore::ContentSecurityPolicy::allowEval const):
(WebCore::ContentSecurityPolicy::allowFrameAncestors const):
(WebCore::ContentSecurityPolicy::allowPluginType const):
(WebCore::ContentSecurityPolicy::allowObjectFromSource const):
(WebCore::ContentSecurityPolicy::allowChildFrameFromSource const):
(WebCore::ContentSecurityPolicy::allowResourceFromSource const):
(WebCore::ContentSecurityPolicy::allowWorkerFromSource const):
(WebCore::ContentSecurityPolicy::allowScriptFromSource const):
(WebCore::ContentSecurityPolicy::allowStyleFromSource const):
(WebCore::ContentSecurityPolicy::allowConnectToSource const):
(WebCore::ContentSecurityPolicy::allowBaseURI const):
(WebCore::ContentSecurityPolicy::allowTrustedTypesPolicy const):
(WebCore::ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup const):
* Source/WebCore/page/csp/ContentSecurityPolicy.h:
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::checkedCompositor):
* Source/WebCore/rendering/RenderView.h:

Canonical link: https://commits.webkit.org/300009@main
